tracee-rules/engine/engine.go

```diff
@@ -107,7 +107,6 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 						}
 					}
 				}
-				engine.inputs.Tracee = nil // TODO: Setting the channel to nil would block any future writes https://play.golang.org/p/jWyYwjNK_NT
```

Comment:
what's the motivation for this change? this line is preventing the containing loop from continuing infinitely and wasting cycles. I'm open to another solution but we need to be mindful of these changes

Reply:
consumeSources is just a consumer of the channel. In this case if the following happens:

1. Engine is running and processing events
2. Producer closes channel but does *not* send `done`
3. Engine will assign `nil` to the input channel
4. Producer will open the channel (recreate) and try to send another event
5. Engine will continue to wait and do nothing

In this case if we don't set `engine.inputs.Tracee = nil` when the producer does step 4. the engine will automatically pick it up and continue on with the newly created channel.

Here's a test if you want to see it in action: https://gist.github.com/simar7/0576c9d5d61c3f6785883b1cf936769f

Reply:
> 4. Producer will open the channel (recreate) and try to send another event

This is impossible. Producer, which is outside `package engine` (in our case `package main`), cannot manipulate `engine.inputs`.

Reply:
I see your point about abstraction. You're right, I'll revert this. I also don't know how to prevent a busy loop within a select waiting on channels but it's something I'm interested in learning more about.

Reply:
thanks. if you want to investigate it further, put a print in the `consumeSources` where it handles the closed channel, then you can "see" the busy loop